src/icesat2_tracks/analysis_db/B06_correct_separate_var.py

```diff
@@ -0,0 +1,773 @@
+
+    B3_v2[bb] = T3
+    Gx_v2[bb] = Gx_k
+
+try:
+    G_angle = xr.open_dataset(load_path_angle + "/B05_" + ID_name + "_angle_pdf.nc")
+
+    font_for_pres()
+
+    Ga_abs = (
+        G_angle.weighted_angle_PDF_smth.isel(angle=G_angle.angle > 0).data
+        + G_angle.weighted_angle_PDF_smth.isel(angle=G_angle.angle < 0).data[:, ::-1]
+    ) / 2
+    Ga_abs = xr.DataArray(
+        data=Ga_abs.T,
+        dims=G_angle.dims,
+        coords=G_angle.isel(angle=G_angle.angle > 0).coords,
+    )
+
+    Ga_abs_front = Ga_abs.isel(x=slice(0, 3))
+    Ga_best = (Ga_abs_front * Ga_abs_front.N_data).sum("x") / Ga_abs_front.N_data.sum(
+        "x"
+    )
+
+    theta = Ga_best.angle[Ga_best.argmax()].data
+    theta_flag = True
+
+    font_for_print()
+    F = M.figure_axis_xy(3, 5, view_scale=0.7)
+
+    plt.subplot(2, 1, 1)
+    plt.pcolor(Ga_abs)
+    plt.xlabel("abs angle")
+    plt.ylabel("x")
+
+    ax = plt.subplot(2, 1, 2)
+    Ga_best.plot()
+    plt.title("angle front " + str(theta * 180 / np.pi), loc="left")
+    ax.axvline(theta, color="red")
+    F.save_light(path=plot_path, name="B06_angle_def")
+except:
+    print("no angle data found, skip angle corretion")
+    theta = 0
+    theta_flag = False
+
+# %%
+lam_p = 2 * np.pi / Gk.k
+lam = lam_p * np.cos(theta)
+
+if theta_flag:
+    k_corrected = 2 * np.pi / lam
+    x_corrected = Gk.x * np.cos(theta)
+else:
+    k_corrected = 2 * np.pi / lam * np.nan
+    x_corrected = Gk.x * np.cos(theta) * np.nan
+
+# spectral save
+G5 = G_gFT_wmean.expand_dims(dim="beam", axis=1)
+G5.coords["beam"] = ["weighted_mean"]
+G5 = G5.assign_coords(N_photons=G5.N_photons)
+G5["N_photons"] = G5["N_photons"].expand_dims("beam")
+G5["N_per_stancil_fraction"] = G5["N_per_stancil_fraction"].expand_dims("beam")
+
+Gk_v2 = xr.merge([Gk, G5])
+
+Gk_v2 = Gk_v2.assign_coords(x_corrected=("x", x_corrected.data)).assign_coords(
+    k_corrected=("k", k_corrected.data)
+)
+
+Gk_v2.attrs["best_guess_incident_angle"] = theta
+
+# save collected spectral data
+Gk_v2.to_netcdf(save_path + "/B06_" + ID_name + "_gFT_k_corrected.nc")
+Gx
+# save real space data
+Gx.to_netcdf(save_path + "/B06_" + ID_name + "_gFT_x_corrected.nc")
+try:
+    io.save_pandas_table(
+        B2_v2, "B06_" + ID_name + "_B06_corrected_resid", save_path
+    )  # all photos but heights adjusted and with distance coordinate
+except:
+    os.remove(save_path + "B06_" + ID_name + "_B06_corrected_resid.h5")
+    io.save_pandas_table(
+        B2_v2, "B06_" + ID_name + "_B06_corrected_resid", save_path
+    )  # all photos but heights adjusted and with distance coordinate
+
+try:
+    io.save_pandas_table(
+        B3_v2, "B06_" + ID_name + "_binned_resid", save_path
+    )  # regridding heights
+except:
+    os.remove(save_path + "B06_" + ID_name + "_binned_resid.h5")
+    io.save_pandas_table(
+        B3_v2, "B06_" + ID_name + "_binned_resid", save_path
+    )  # regridding heights
```

Comment:
This can be refactored to something like this but it's tricky to tell it works as it should. To be safe better wait for test suite.
```suggestion
def save_table(data, tablename, save_path):
    try:
        io.save_pandas_table(data, tablename, save_path)
    except Exception as e:
        tabletoremove = save_path + tablename + ".h5"
        print(e, f"Failed to save table. Removing {tabletoremove} and re-trying..")
        os.remove(tabletoremove)
        io.save_pandas_table(data, tablename, save_path)

B06_ID_name = "B06_" + ID_name
table_names = [B06_ID_name + suffix for suffix in ["_B06_corrected_resid", "_binned_resid"]]
data = [B2_v2, B3_v2]

for tablename, data in zip(table_names, data):
    save_table(data, tablename, save_path)
```

Reply:
I like this suggestion, pro-coding style. I dont see a reason why not to include it. I just did and it works as expected.